include/Graph/Graph.hpp

```diff
@@ -0,0 +1,71 @@
+#pragma once
+#include <iostream>
+#include <stdexcept>
+#include <string>
+#include <vector>
+
+class Node {
```

Comment:
please use adjecency list with std::vector, pointers are potential problem for memory and control of meomry